### PR TITLE
stat: add BivariateMoment

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
If anyone is wondering why this is useful, this computes the mu terms in https://stats.stackexchange.com/questions/48366/standard-errors-for-covariance-estimate-in-r

File: stat/stat.go [hunk 1]
```diff
@@ -876,13 +911,17 @@ func Moment(moment float64, x, weights []float64) float64 {
 		}
 		return m / float64(len(x))
 	}
+	if len(x) != len(weights) {
```
Comment (kortschak) [inline stat/stat.go]: This should really happen in another PR.

File: stat/stat.go [hunk 2]
```diff
@@ -876,13 +911,17 @@ func Moment(moment float64, x, weights []float64) float64 {
...
+		panic("stat: slice length mismatch")
+	}
 	var (
 		m          float64
 		sumWeights float64
 	)
 	for i, v := range x {
-		m += weights[i] * math.Pow(v-mean, moment)
-		sumWeights += weights[i]
+		w := weights[i]
```
Comment (kortschak) [inline stat/stat.go]: Same here.

File: stat/stat.go [hunk 3]
```diff
@@ -862,6 +862,41 @@ func Mode(x, weights []float64) (val float64, count float64) {
 	return max, maxCount
 }
 
+// MixedMoment computes the weighted mixed moment between the samples x and y.
```
Comment (kortschak) [inline stat/stat.go]: `BivariateMixedMoment` in case we want higher order mixed moments?

**Comment by btracey** [comment]
I changed to `BivariateMoment` instead of `BivariateMixedMoment`

**Comment by btracey** [comment]
I'm opening an issue for the example. The variance of the covariance as liked above is a good example, but I need to test that it's correct.

**Pull request merged**